cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/TransactionalFunctionExecutor.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.spi.authorization;
+
+import co.cask.cdap.api.Transactional;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.tephra.TransactionFailureException;
+
+/**
+ * A {@link Transactional} that allows users to return a value of the specified type from the transaction.
+ */
+public interface TransactionalFunctionExecutor extends Transactional {
```

Comment:
Correction, we don't need this anymore, this can be implemented in the extension itself. Removed the ``TxCallable`` as well as ``TransactionalFunctionExecutor``